clang/lib/CodeGen/CGCall.cpp

```diff
@@ -753,13 +753,7 @@ CodeGenTypes::arrangeCall(const CGFunctionInfo &signature,
                                  signature.getRequiredArgs());
 }
 
-namespace clang {
-namespace CodeGen {
-void computeSPIRKernelABIInfo(CodeGenModule &CGM, CGFunctionInfo &FI);
-}
-}
```

Comment:
I'm not sure I understand the motivation of moving this. Why is it more desirable to have this in the header? A search shows only 3 uses of the symbol in the tree (https://github.com/search?q=repo%3Allvm%2Fllvm-project%20computeSPIRKernelABIInfo&type=code). Given that additional headers are required to declare this in `CGCall.h`, I'd say we may prefer the forward declaration. I'm basing that on https://llvm.org/docs/CodingStandards.html#include-as-little-as-possible, so unless there's a good reason to deviate, I'd suggest dropping this part. It also doesn't seem related to the rest of the patch.